Sources/Workspace/Workspace.swift

```diff
@@ -722,7 +722,8 @@ extension  Workspace {
             guard !diagnostics.hasErrors else { break resolve }
 
             // Update the checkouts with dependency resolution result.
-            try updateCheckouts(with: result)
+            updateCheckouts(with: result, diagnostics: diagnostics)
+            guard !diagnostics.hasErrors else { break resolve }
```

Comment:
We can move the do block down now because there is nothing throwing untill pinall anymore